extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaRecordSupplier.java

```diff
@@ -55,20 +57,29 @@
 import java.util.Properties;
 import java.util.Set;
 import java.util.concurrent.Callable;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 
-public class KafkaRecordSupplier implements RecordSupplier<Integer, Long, KafkaRecordEntity>
+public class KafkaRecordSupplier implements RecordSupplier<KafkaTopicPartition, Long, KafkaRecordEntity>
 {
   private final KafkaConsumer<byte[], byte[]> consumer;
   private boolean closed;
 
+  private boolean multiTopic;
+
+  // Store the stream information when partitions get assigned. This is required because the consumer does not
+  // know about the parent stream which could be a list of topics.
+  private String stream;
```

Comment:
Does this need to be `volatile`?
Also, maybe convert the comment into a javadoc. It would probably be nice to call out that this would have the same value as the `topic` field in the supervisor spec.

Reply:
don't think that is needed. its updated only during the assign and other methods can't be called till we have done an assignment.